pulsar-jms/src/main/java/com/datastax/oss/pulsar/jms/PulsarJMSProducer.java

```diff
@@ -192,9 +192,6 @@ private void getProducerAndSend(Destination destination, Message message) throws
       message.setObjectProperty(prop.getKey(), prop.getValue());
     }
     message.setJMSPriority(priority);
-    message.setJMSCorrelationIDAsBytes(correlationID);
```

Comment:
https://jakarta.ee/specifications/messaging/3.0/apidocs/jakarta/jms/jmsproducer

> message headers set using these methods will override any message headers that have been set directly on the message

this seems to imply that it is possible to set the correlationid on the message

According to the specs I would code it this way:

if (!message.hasCorrelationId()){
 message.setJMSCorrelationIDAsBytes(correlationID);
}